### PR TITLE
8293230: x86_64: Move AES and GHASH stub definitions into separate source files

### PR DESCRIPTION
**Pull request by iwanowww**
Move code related to AES and GHASH stubs from `stubGenerator_x86_64.cpp` into `stubGenerator_x86_64_aes.cpp` and `stubGenerator_x86_64_ghash.cpp` respectively.

Testing: hs-tier1 - hs-tier4

/cc hotspot

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293230](https://bugs.openjdk.org/browse/JDK-8293230): x86_64: Move AES and GHASH stub definitions into separate source files

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10124/head:pull/10124` \
`$ git checkout pull/10124`

Update a local copy of the PR: \
`$ git checkout pull/10124` \
`$ git pull https://git.openjdk.org/jdk pull/10124/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10124`

View PR using the GUI difftool: \
`$ git pr show -t 10124`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10124.diff">https://git.openjdk.org/jdk/pull/10124.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/x86/stubGenerator_x86_64.hpp [hunk 1]
```diff
@@ -245,22 +245,38 @@ class StubGenerator: public StubCodeGenerator {
...
+  address generate_upper_word_mask();
+  address generate_shuffle_byte_flip_mask();
+  address generate_pshuffle_byte_flip_mask();
+
+
+  // AES intrinsic stubs
+
+  enum {
+    AESBlockSize = 16
+  };
```
Comment (vnkozlov) [inline src/hotspot/cpu/x86/stubGenerator_x86_64.hpp:279]: Why it is not in new `aes` file?

**Review by vnkozlov — COMMENTED** [review]
What about stubGenerator_x86_32.cpp? It also has AES intrinsic code.

**Comment by iwanowww** [comment]
> What about stubGenerator_x86_32.cpp? It also has AES intrinsic code.

This PR is self-contained and deliberately focused on x86-64. I don't have ability to test x86-32 changes, but if anybody wants to contribute a patch covering `stubGenerator_x86_32.cpp`, I'll be more than happy to incorporate it.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.hpp (hunk 1 above)
Comment (iwanowww) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.hpp]: Fixed.

**Comment by vnkozlov** [comment]
> > What about stubGenerator_x86_32.cpp? It also has AES intrinsic code.
> 
> This PR is self-contained and deliberately focused on x86-64. I don't have ability to test x86-32 changes, but if anybody wants to contribute a patch covering `stubGenerator_x86_32.cpp`, I'll be more than happy to incorporate it.

I thought `macroAssembler_x86_aes.cpp` had both implementations. I was wrong.

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by vnkozlov — COMMENTED** [review]
@iwanowww I noticed that your repo do not have GitHub actions testing. Please, switch it on.

**Comment by iwanowww** [comment]
> I noticed that your repo do not have GitHub actions testing. Please, switch it on.

@vnkozlov I don't know how it is expected to work, but right now I see there's GHA testing going on.

**Comment by vnkozlov** [comment]
> > I noticed that your repo do not have GitHub actions testing. Please, switch it on.
> 
> @vnkozlov I don't know how it is expected to work, but right now I see there's GHA testing going on.

I see it now. Yesterday it was not present in this and an other your PR.

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by iwanowww** [comment]
Thanks for the reviews, Vladimir and Tobias.

/integrate

**Pull request closed without merging**